### ISSUE TITLE
1.6  Runaway PHP-fpm on certain actions related to plugins.

### ISSUE THREAD

**Issue by tailz303** [FOGProject/fogproject#668]
1.6 is almost perfect with its beautiful interface, pagination, and (new to us) user access controls. We are running into a couple issues where we can enable the access control plugin, we can create roles and edit role rules but we cannot delete them. anytime we try to initiate a deletion the page hangs until PHP-fpm runs out of memory and then it logs it in ApacheDIR/error.log. If you up the limit it will just continue to run away until it hits that limit. same thing happens when trying to install the LDAP plug in, something is broken and that is not installable. we tried doing similar things on our 1.5.10 servers and they do not have the crash.

**To Reproduce**
Install access control plugin
go to access control plugin
try to delete roles or rules
wait for general error

**Expected behavior**
For it to delete the selection 

**Software (please complete the following information):**
 - FOG version 1.6-beta
 - FOS kernel unsure-not available at the moment-latest
 - OS: Ubuntu server 24.04

**Comment by Alterak** [comment]
I have the same problem when deleting an existing image-)
I decided to perform a clean install, but now I can't create an image-) I'll have to go back to version 1.5

**Comment by tailz303** [comment]
I'm wondering if any of the smarter people on the code base of this project know at least where the issue is so I can try to fix it? My job requires fog for 20k machines and lack of pagination on 1.5 means that I need to look at 1.6 and centralization as we call it internally.

Currently we have 7 groups of 3 servers running with 2 as slave nodes for each group, and a bigger server for all of our elementary schools with 30 locations on it. This is mostly a pain because of the number of failure points. 1.6 has been solid except for a few minor things which is honestly impressive and I appreciate everyone that has contributed.

**Comment by darksidemilk** [comment]
Can you post the relevant lines, or all if you're unsure, of the log files in question.
If on redhat based linux
/var/log/php-fpm/www-error.log I believe. That will help trace this further

**Comment by tailz303** [comment]
Only useful log I was able to find was the apache error log which had to be enabled in the configuration since its not be default. The following is all errors from when apache is restarted all the way to when I try to delete a role from within the access control plugin. I am using php8.1 if that helps narrow it down. I checked the fpm log and it wasn't anything seemingly relevant. I can provide anything else needed, just let me know.

[fog-1.6-error.log.txt](https://github.com/user-attachments/files/18264327/fog-1.6-error.log.txt)
`tech@fog-01-proxmox:/etc/php$ tail -f /var/log/apache2/error.log
[Fri Dec 27 23:31:47.719682 2024] [ssl:warn] [pid 444224] AH01909: 10.255.6.14:443:0 server certificate does NOT include an ID which matches the server name
[Fri Dec 27 23:31:47.763083 2024] [ssl:warn] [pid 444225] AH01909: 10.255.6.14:443:0 server certificate does NOT include an ID which matches the server name
[Fri Dec 27 23:31:47.768303 2024] [mpm_prefork:notice] [pid 444225] AH00163: Apache/2.4.52 (Ubuntu) OpenSSL/3.0.2 configured -- resuming normal operations
[Fri Dec 27 23:31:47.768358 2024] [core:notice] [pid 444225] AH00094: Command line: '/usr/sbin/apache2'
[Fri Dec 27 23:32:46.525385 2024] [proxy_fcgi:error] [pid 444230] [client 205.126.67.66:57168] AH01071: Got error 'PHP message: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 450560 bytes) in /var/www/html/fog/lib/fog/fogbase.class.php on line 2447', referer: http://10.255.6.14/fog/management/index.php?node=accesscontrol&sub=list`

**Comment by darksidemilk** [comment]
> I have the same problem when deleting an existing image-) I decided to perform a clean install, but now I can't create an image-) I'll have to go back to version 1.5

Where are you getting stuck? Are you unable to capture an image or unable to create in the gui? Are you getting any errors? @Alterak

**Comment by darksidemilk** [comment]
@tailz303 Sorry for the delay, but back to normal work after the holidays now. 
Looking at that error it's running out of allowed memory in php. I believe the default is like 128 MB.

On your fog server terminal, try editing `/etc/php.ini` and thange `memory_limit = 128M` to a higher number such as 256, 512, or 1024 or whatever you desire.
In your fog server web gui under FOG Configuration also increase the FOG_MEMORY_LIMIT setting to the same.

Then you may need to restart php and apache (something like `systemctl restart php-fpm & systemctl restart apache` I don't remember the service names in ubunutu) or you can just reboot the server for good measure.

Then try it again and report back.

@mastacontrola can probably help in case there's a spot I missed for increasing the memory.

Based on the scope of your servers you've previously described, it might just need a bit more memory. There could also be a memory leak and maybe with enough memory the error.log will actually show more trace on where this is actually happening.

**Comment by darksidemilk** [comment]
I also just found a recentish pull request that is in the stable branch but not in the 1.6 branch that relates to the access control plugin and php 8 compatibility https://github.com/FOGProject/fogproject/pull/647 if increasing the memory doesn't do the trick we can try applying the same changes to the working-1.6 branch.

**Comment by darksidemilk** [comment]
I just re-read your original post and see that you already tried increasing the limit...my bad.
I was able to recreate the error and it only happens when deleting from the list of access rules, if you go into the rule and delete from the individual rule it works fine.

**Comment by tailz303** [comment]
Yeah I think we had the mem limit at 2G for a bit but it just kept filling up so we restarted fpm after it used 512M as it was taking forever to crash and it breaks the whole server when that happens. I'm relatively new to GitHub so if you want me to try something on our server/update from a different branch just let me know and I'll give it a go!

**Comment by darksidemilk** [comment]
It looks like there's an infinite loop happening when deleting multiple rules when it's also trying to find and delete any associations to that rule. I'm seeing if I can find a solution. I don't actually use this plugin so I can only test so much, but I was at least able to recreate the issue on my dev fog server

**Comment by darksidemilk** [comment]
@tailz303 with the help of @mastacontrola  we got the problem debugged and fixed. If you update to the latest 1.6.
After you confirm that fixes it for you too we can close this one out.

**Comment by tailz303** [comment]
Okay I ran the installfog.sh and this is the version it claims so hopefully this is correct " Version: 1.6.0-beta.2147 Installer/Updater"

This appears to have fixed the issue deleting roles, rule associations, and just for sanity sake I created and deleted an image successfully. Thank you so much for fixing that!

**Comment by darksidemilk** [comment]
@tailz303 We're happy to help. Thank you for your patience and for reporting the issue. If you have any other feedback on 1.6 we're happy to hear it.

**Comment by Alterak** [comment]
> @tailz303 with the help of @mastacontrola we got the problem debugged and fixed. If you update to the latest 1.6. After you confirm that fixes it for you too we can close this one out.

Hi, I'll try updating the version too, maybe it will also solve my capture problem. If the problem persists, I will create a separate new request on GitHub so as not to confuse you in this topic.
